src/utils/appDataClient.ts

```diff
@@ -24,7 +25,27 @@ import { fromNow } from './timeUtils'
 import * as Logger from './logger'
 
 const MAX = 100000
-const client = new Client()
+let client = new Client(Network.stakejoy)
+
+// Always read pending txns from helium
+const pendingTxnsClient = new Client(Network.production)
+
+const compareNetwork = (network: string) => {
+  return (
+    (network === 'stakejoy' && client.network === Network.stakejoy) ||
+    (network === 'helium' && client.network === Network.production)
+  )
+}
+
+export const updateClient = (nextNetwork: string) => {
+  const isSame = compareNetwork(nextNetwork)
+  if (!isSame) {
+    const network =
+      nextNetwork === 'helium' ? Network.production : Network.stakejoy
+    client = new Client(network)
```

Comment:
Could this get reinitialized while things are already partly paged and throw off the cursor? Or do we just call this and everything in the app refreshes?

Reply:
I had the same concern. A couple of lines down I re-init the txn fetchers and after thorough testing, it all seems to work. Could you test when you have a chance? The wallet API is updated now.